src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/group/util/GroupUtil.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 Exactpro (Exactpro Systems Limited)
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.common.schema.message.impl.rabbitmq.group.util
+
+import com.exactpro.th2.common.grpc.AnyMessage
+import com.exactpro.th2.common.grpc.MessageGroup
+import com.exactpro.th2.common.grpc.MessageGroupBatch
+
+/**
+ * Transfers messages that match the [filterFunction] from current group to new group in [batchBuilder].
+ *
+ * If all messages are filtered no group will be added.
+ *
+ * If all messages match the filter the current group will be added into [batchBuilder] as is
+ */
+internal fun MessageGroup.filterMessagesToNewGroup(
+    batchBuilder: MessageGroupBatch.Builder,
+    filterFunction: (AnyMessage) -> Boolean,
+): List<AnyMessage> {
+    class Match(val message: AnyMessage, val matched: Boolean)
+
+    val (matched, dropped) = messagesList.asSequence()
+        .map { Match(it, filterFunction(it)) }
+        .partition { it.matched }
+    if (messagesCount == matched.size) {
+        batchBuilder.addGroups(this)
+        return emptyList()
+    }
+    if (matched.isEmpty()) {
+        return messagesList
+    }
+    val newGroup = batchBuilder.addGroupsBuilder()
+    matched.forEach { newGroup.addMessages(it.message) }
+    return dropped.map { it.message }
```

Comment:
```suggestion
    val (matched, dropped) = messagesList.partition { filterFunction(it) }

    when {
        dropped.isEmpty() -> batchBuilder.addGroups(this)
        else -> batchBuilder.addGroupsBuilder().addAllMessages(matched)
    }

    return dropped
```

Reply:
Thank you, my bad. I don't know why I have added the extra wrapping instead of simple partitioning by filter.
But I want to keep all created lists local if it is possible so I will change it a bit

Reply:
> I want to keep all created lists local

Why? If you fear that `dropped` list could be modified from Java you can wrap it in `Collections.unmodifiableList`. Otherwise I see no reason in doing so.